### PR TITLE
feat: optimize interactive block processing logic in context_v2

### PR DESCRIPTION
**Pull request by aichy126**
# Summary

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

## Summary by CodeRabbit

* **Bug Fixes**
  * Enhanced block context processing to safely handle boundary edge cases and expanded support for markdown interaction syntax expansion.

* **Dependencies**
  * Updated markdown-flow library to version 0.2.81.

* **Tests**
  * Added test coverage for interaction block transformation and expansion.

### CODE REVIEW

File: src/api/flaskr/service/learn/context_v2.py [hunk 1]
```diff
@@ -491,11 +491,10 @@ def build_context_from_blocks(
         current_app.logger.info(f"build_context_from_blocks variables: {variables}")
 
         for generated_block in blocks:
-            if (
-                generated_block.type == BLOCK_TYPE_MDCONTENT_VALUE
-                and generated_block.position < len(block_list)
-            ):
-                block = block_list[generated_block.position]
+            if generated_block.position >= len(block_list):
+                continue
```
[the remark this answers is not in the document]
Comment (aichy126) [reply src/api/flaskr/service/learn/context_v2.py]: Good catch — adopted in 1f1dd7c2. Added an explicit `generated_block.position < 0` check so negative positions are skipped instead of wrapping around to the wrong block via Python negative indexing.
